### ISSUE TITLE
expand all folders during file search

### ISSUE THREAD

**Issue by RonConescu** [microsoft/vscode#139223]
Issue Type: <b>Feature Request</b>

Folks,

Please add this feature:

When I start typing in the file list, creating a search for certain file/folder names, please (a) find all those items in all available folders, and (b) expand all appropriate parent folders to show those items. Or even better (c) hide all items that don't match.

Reason: I'm learning a new code base. There are many many MANY files. I often have to search for files by name, because many filenames don't match the contents (e.g., a file named "one.py" that declares a class "Whatever" instead of a class named "One"). These files are located in unintuitive-to-me locations, through a very complex set of include paths, partly because they all get precompiled and moved when they get to their target environment on a different machine. So I don't know where the files are, or if they exist.

As things are now, VSCode only highlights the files that are ALREADY VISIBLE which match my search. Unfortunately, that isn't what I need most, because at any given moment, most of the places to look for files are hidden -- there are thousands more places to search.

So I would love to be able to ask VSCode to hunt through all those places and show me the matches.

Thank you so much for considering it!

Ron Conescu

VS Code version: Code 1.63.1 (fe719cd3e5825bf14e14182fddeb88ee8daf044f, 2021-12-14T02:02:02.183Z)
OS version: Linux x64 5.11.0-43-generic
Restricted Mode: No

**Comment by gjsjohnmurray** [comment]
When you hover the mouse over the widget that displays the characters you typed you get a button to switch to filter mode instead of highlighting mode. There's also a setting for which mode is the default. 

There is another issue open about expanding folders before searching the tree.

**Comment by RonConescu** [comment]
@gjsjohnmurray  -- wow! Thanks for the fast reply!

I think I'm looking for something in the middle of all these versions of the situation. The current "filter" mode seems to only filter the VISIBLE items. I can see how "expand before searching the tree" might get partway there, but I don't think that would be quite the same thing. If I expand all the folders, I'd see thousands of folders and tens of thousands of files, most of which would be irrelevant.

I'd imagine and hope that VSCode would do this:
* find matching files in the collapsed folders
* expand only those parent folders, so I can see all the matching files items in their context
* if I went into "filter" mode from THAT view, we'd get the current "filter" behavior
* if I cancelled out of the search, it'd either revert to my previous view or keep the current list of files and folders expanded

Am I making any sense?
